crates/spicepod/src/component/worker.rs

```diff
@@ -0,0 +1,122 @@
+
+use super::{Nameable, WithDependsOn};
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+use serde::de::{self, Deserializer};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
+#[serde(rename_all = "kebab-case")]
+pub enum RoutingStrategy {
+    #[default]
+    Fallback,
+    RoundRobin,
+    Weighted,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+#[serde(untagged)]
+pub enum RouterConfig {
+    Fallback { from: String, order: u32 },
+    Weighted { from: String, weight: u32 },
+    RoundRobin { from: String }, // Must be last for deserialization.
+}
+
+impl RouterConfig {
+    #[must_use]
+    pub fn from(&self) -> String {
+        match self {
+            RouterConfig::Fallback { from, .. }
+            | RouterConfig::RoundRobin { from }
+            | RouterConfig::Weighted { from, .. } => from.clone(),
+        }
+    }
+}
+
+impl From<&RouterConfig> for RoutingStrategy {
+    fn from(value: &RouterConfig) -> Self {
+        match value {
+            RouterConfig::Fallback { .. } => RoutingStrategy::Fallback,
+            RouterConfig::RoundRobin { .. } => RoutingStrategy::RoundRobin,
+            RouterConfig::Weighted { .. } => RoutingStrategy::Weighted,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+pub struct Worker {
+    pub name: String,
+
+    pub description: Option<String>,
+
+    #[serde(default, skip_serializing_if = "HashMap::is_empty")]
+    pub params: HashMap<String, Value>,
+
+    #[serde(
+        default,
+        skip_serializing_if = "Vec::is_empty",
+        deserialize_with = "deserialize_router_configs"
+    )]
+    pub models: Vec<RouterConfig>,
+}
+
+impl Nameable for Worker {
+    fn name(&self) -> &str {
+        &self.name
+    }
+}
+
+impl WithDependsOn<Worker> for Worker {
+    fn depends_on(&self, _depends_on: &[String]) -> Worker {
+        Worker {
+            name: self.name.clone(),
+            description: self.description.clone(),
+            params: self.params.clone(),
+            models: self.models.clone(),
+        }
+    }
+}
+
+fn deserialize_router_configs<'de, D>(deserializer: D) -> Result<Vec<RouterConfig>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let configs = Vec::<RouterConfig>::deserialize(deserializer)?;
+
+    // If there is at least one element, check they are all the same variant.
+    let Some(first) = configs.first() else {
+        return Err(de::Error::custom(
+            "Worker requires at least one model specified in `.models`",
+        ));
+    };
+
+    let strategy: RoutingStrategy = first.into();
+
+    if configs.iter().any(|c| RoutingStrategy::from(c) != strategy) {
+        return Err(de::Error::custom(
+            "All `.models` must be the same format, but are not.",
+        ));
+    }
```

Comment:
```suggestion
    if configs.iter().any(|c| c != first) {
        return Err(de::Error::custom(
            "All `.models` must be the same format, but are not.",
        ));
    }
```

Couldn't we replace this with an `Eq` implementation on `RouterConfig`? I'm not exactly sure why the `RoutingStrategy` enum exists when `RouterConfig` is already an enum.

Reply:
`// If there is at least one element, check they are all the same variant.` ok, this makes more sense now.